monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,153 @@ def get_params(self, image_size):
+            Defaults to ``255``.
+        tile_filter_mode: [min, max, None] filter tiles. If total number of tiles is more then tile_size,
+            then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for None) subset
+            Defaults to ``min`` (which assumes background is white, high value)
+
+    """
+
+    def __init__(
+        self,
+        tile_count: int = None,
+        tile_size: int = 256,
+        tile_step: int = None,
+        tile_all: bool = False,
+        tile_random_offset: bool = False,
+        tile_pad_full: bool = False,
+        tile_background_val: int = 255,
+        tile_filter_mode: str = "min",  # None, min, max
+    ):
+        self.tile_count = tile_count
+        self.tile_size = tile_size
+        self.tile_step = tile_step
+        self.tile_all = tile_all
+        self.tile_random_offset = tile_random_offset
+        self.tile_pad_full = tile_pad_full
+        self.tile_background_val = tile_background_val
+        self.tile_filter_mode = tile_filter_mode
+
+        if self.tile_step is None:
+            self.tile_step = self.tile_size  # non-overlapping grid
+
+        self.random_offset = None
+        self.random_idxs = None
+
+    def randomize(self, img_size: Sequence[int] = None) -> None:
+
+        c, h, w = img_size
+
+        if img_size is not None:
+            pad_h = h % self.tile_size
+            pad_w = w % self.tile_size
+            if pad_h > 0 and pad_w > 0:
+                self.random_offset = (self.R.randint(pad_h), self.R.randint(pad_w))
+                h = h - self.random_offset[0]
+                w = w - self.random_offset[1]
+            else:
+                self.random_offset = None
+
+        if self.tile_pad_full:
+            pad_h = (self.tile_size - h % self.tile_size) % self.tile_size
+            pad_w = (self.tile_size - w % self.tile_size) % self.tile_size
+            h = h + pad_h
+            w = w + pad_w
+
+        h_n = (h - self.tile_size + self.tile_step) // self.tile_step
+        w_n = (w - self.tile_size + self.tile_step) // self.tile_step
+        tile_total = h_n * w_n
+
+        if tile_total > self.tile_count:
+            self.random_idxs = self.R.choice(range(tile_total), self.tile_count, replace=False)
+        else:
+            self.random_idxs = None
+
+    def __call__(self, image: np.ndarray) -> np.ndarray:
+
+        # add random offset
+        self.randomize(img_size=image.shape)
+
+        if self.tile_random_offset and self.random_offset is not None:
+            image = image[:, self.random_offset[0] :, self.random_offset[1] :]
+
+        # pad to full size, divisible by tile_size
+        if self.tile_pad_full:
+            c, h, w = image.shape
+            pad_h = (self.tile_size - h % self.tile_size) % self.tile_size
+            pad_w = (self.tile_size - w % self.tile_size) % self.tile_size
+            image = np.pad(
```

Comment:
I would suggest to use MONAI `Pad` transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L72

Thanks.

Reply:
I don't see any reason to use Pad (from monai) here.  A simple np.pad does the job and is concise.